dbms/src/Common/HashTable/HashTable.h

```diff
@@ -224,7 +224,11 @@ struct ZeroValueStorage<true, Cell>
 
 public:
     bool hasZero() const { return has_zero; }
-    void setHasZero() { has_zero = true; }
+    void setHasZero()
+    {
+        has_zero = true;
+        zero_value_storage = {};
```

Comment:
I think we should call the Cell constructor here, like we do for non-zero Cells. We already have Cells with non-trivial constructors, like ClearableHashTableCell.

Reply:
ClearableHashTableCell  doesn't use zero storage

https://github.com/yandex/ClickHouse/blob/master/dbms/src/Common/HashTable/ClearableHashSet.h#L45

Reply:
And I don't think constructing an std::aligned_storage_t as Cell even works...